linode/resource_linode_instance.go

```diff
@@ -1160,16 +1160,44 @@ func resourceLinodeInstanceUpdate(d *schema.ResourceData, meta interface{}) erro
 		d.Partial(false)
 	}
 
-	if d.HasChange("type") {
-		if err = changeInstanceType(&client, instance, d.Get("type").(string), d); err != nil {
-			return err
-		}
-		d.Set("type", d.Get("type").(string))
-	}
+	var rebootInstance bool
+	var diskIDLabelMap map[string]int
 
 	tfDisksOld, tfDisksNew := d.GetChange("disk")
 
-	rebootInstance, diskIDLabelMap, err := updateInstanceDisks(client, d, *instance, tfDisksOld, tfDisksNew)
+	tfDisksOldInterface := tfDisksOld.([]interface{})
+	tfDisksNewInterface := tfDisksNew.([]interface{})
+
+	oldDiskSize := 0
+	newDiskSize := 0
+
+	// Get total amount of disk usage before & after
+	for _, disk := range tfDisksOldInterface {
+		oldDiskSize += disk.(map[string]interface{})["size"].(int)
+	}
+
+	for _, disk := range tfDisksNewInterface {
+		newDiskSize += disk.(map[string]interface{})["size"].(int)
+	}
+
+	if newDiskSize > oldDiskSize {
```

Comment:
Why is the order of steps dependent on the size of the two disks? Is there any reason we can't just choose one of these orderings?

Reply:
imagining the following change:
1 GB Linode with 25 GB SSD -> 2 GB Linode with 40 GB SSD
We need to change the instance size here first, and then the disk size, or else the disk will be too large for the instance.

Going the other way:
2 GB instance with 50 GB disk -> 1 GB instance with 20 GB disk
We need to shrink the disk first, or else we will not be able to get all 50 GB on the smaller linode instance type.

another case:
2 GB w/ 10 GB ssd -> 1 GB with 25 GB SSD:
order does not matter

2 GB w/ 25 -> 1 GB with 50 GB:
produce error

.. is that sensible?